mlir/include/mlir/Dialect/XeGPU/IR/XeGPUTypes.td

```diff
@@ -189,6 +189,7 @@ def XeGPU_TensorDesc: XeGPUTypeDef<"TensorDesc", "tensor_desc",
   let genVerifyDecl = 1;
 }
 
+def XeGPU_TensorDescOrMemRef : AnyTypeOf<[XeGPU_TensorDesc,Non0RankedMemRefOf<[XeGPU_ScalarType]>, UI64]>;
```

Comment:
First, unrelated change to the PR. Could it be done separately so the offset addition is not blocked?

Second, why is it now allowed? What are the semantics? Why `load_nd` is now different?
It's basically undefined as is.

Reply:
> What do you mean by "basically undefined as is"?

Looking at the current dialect docs, I have no idea how to interpret the new load memref source. If I have a heavily strided subview, then what happens etc. The descriptor ops provide a lot more semantics.
Now, the above puts a lot of valuable context and sense into the changes and the intended semantics.

Could you update the PR's description/title as the scope of the changes is quite larger than just the offset? And, more importantly, also update the ops' documentation to contain this information?

Reply:
To rephrase a bit, the overall direction sounds good.
However, looking from an outsider's perspective, it's currently hard to derive all these decisions and intensions from the existing upstream docs. With some updates, it should be much easier to understand and follow the new/current design.

Reply:
changed